bricks/flutter_clean_base/__brick__/{{project_name.snakeCase()}}/lib/core/arch/widget/common/theme_switcher.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:flutter/material.dart';
+
+class ThemeModeSwitcher extends StatefulWidget {
+  final Widget Function(BuildContext, ThemeMode, Widget?) builder;
+  final ThemeMode initialThemeMode;
+
+  const ThemeModeSwitcher({
+    required this.builder,
+    this.initialThemeMode = ThemeMode.system,
+    super.key,
+  });
+
+  @override
+  State<ThemeModeSwitcher> createState() => _ThemeModeSwitcherState();
+}
+
+class _ThemeModeSwitcherState extends State<ThemeModeSwitcher> {
+  late final ValueNotifier<ThemeMode> themeNotifier;
```

Comment:
late final ValueNotifier<ThemeMode> _themeNotifier = ValueNotifier(widget.initialThemeMode) would be better